charmverse/app.charmverse.io
# Proposal permissions

Provide proposal permissions engine

Testing
1: Create a new workspace
- You should have default categories all set to full access for the space

2: Create a new category
- The category should have full access for the space

3: Filtering permissions
- Remove permissions from a category
- Create a proposal
- Move it to discussion stage
- Log in with separate user
- Category should be invisible
- Proposal should be invisible

4: Restricting interactions
- As admin: Add space / view comment
- Create discussion stage proposal in category
- Switch to other user
- User should be able to comment
- Switch back to admin and move proposal to vote active stage
- Switch back to user
- User should not be able to vote
- Switch back to admin; upgrade category permissions to view comment vote
- Switch back to user; they should now be able to vote

5: Make page public
- Make any proposal public via share settings
- Open the page. User should be able to view the proposal
